yente/search/indexer.py

```diff
@@ -162,18 +159,24 @@ async def index_entities(
         if next_index not in aliases:
             log.warn("Deleting partial index", index=next_index)
             await provider.delete_index(next_index)
+        if not force:
+            return await index_entities(provider, dataset, force=True)
         return False
 
     await provider.refresh(index=next_index)
     dataset_prefix = construct_index_name(dataset.name)
-    # FIXME: we're not actually deleting old indexes here any more!
     await provider.rollover_index(
         alias,
         next_index,
         prefix=dataset_prefix,
     )
+    # Delete any old indices that are no longer aliased to the current index.
+    for index in await provider.get_all_indices():
+        if index.startswith(dataset_prefix) and index != next_index:
+            log.info("Deleting old index", index=index)
+            await provider.delete_index(index)
```

Comment:
I think we have that covered in the cleanup function in the same module - it's just moved location.

Reply:
Cool, I'll just delete the todo then.